.github/workflows/test.yml

```diff
@@ -24,3 +32,41 @@ jobs:
 
     - name: "nix-build"
       run: nix-build --max-jobs 4 -A all-systems-go
+
+    - name: Calculate performance delta
+      if: runner.os == 'Linux' && github.event_name == 'pull_request'
+      run: |
+        from="$( git merge-base origin/${{ github.base_ref}} HEAD )"
+        to="${{ github.event.pull_request.head.sha }}"
+        echo "Comparing changes from $from to $to"
+        nix-build --max-jobs 4 perf-delta.nix -o perf-delta.txt \
+          --argstr ref HEAD \
+          --argstr from "$from" \
+          --argstr to "$to"
+
+    - name: Read performance delta
+      if: runner.os == 'Linux' && github.event_name == 'pull_request'
+      id: perf
+      uses: juliangruber/read-file-action@v1
+      with:
+        path: ./perf-delta.txt
+
+    - name: Find performance comment
+      if: runner.os == 'Linux' && github.event_name == 'pull_request'
+      uses: peter-evans/find-comment@v1
+      id: fc
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        comment-author: 'github-actions[bot]'
+        body-includes: '<!-- perf comment -->'
```

Comment:
So this updates an existing perf comment, but where do we create a comment with this line for the first time?

I think adding a new comment might also be useful to see changes with commits.

Reply:
This Github action creates a commit if there is none, or updates one if there is one – no change in behavior here (if everything works as intended)